### ISSUE TITLE
Lehe solver single electron under plane wave pulse simulation differs from Yee solver and analytical result

### ISSUE THREAD

**Issue by tomaslevyy** [ComputationalRadiationPhysics/picongpu#3726]
Same simulation of a single electron under a plane wave laser pulse (A0=5, 800 nm, RAMP_INIT = 14.) show diferent results between Lehe and Yee methods, beeing the Yee method in accordance with analytic results [qucosa 102681].
The x-axis amplitude of oscilation at its maximum should be (analitically): A0*wavelenght/(pi)=1.27 micrometers.
maximum x-momentum should be: A0=5 [m*gamma*beta] (this is shown unitless in the right vertical axis, momentum plotted in green discontinuous lines)
maximum y-momuentum should be: A0^2/2=12.5

 Here I show with Yee method:

![plot1](https://user-images.githubusercontent.com/87823051/129391374-049aec82-52e6-4f12-993f-79aaf84a6d09.png)
![plot2](https://user-images.githubusercontent.com/87823051/129391376-0fa1ebf1-6dd4-429a-9a36-0c9cc8705710.png)
![plot3](https://user-images.githubusercontent.com/87823051/129391377-a743e637-719f-43f9-a6c4-259660706883.png)
![plot4](https://user-images.githubusercontent.com/87823051/129391378-6b45026d-83c8-42d3-b15b-bfa935aab5ba.png)

Here I show with Lehe method:

![plota](https://user-images.githubusercontent.com/87823051/129391475-f2c98cd7-5576-493d-a73e-4fe8f3289b42.png)
![plotb](https://user-images.githubusercontent.com/87823051/129391478-d3dcc791-aab8-4dc4-91c0-dc67426b0f86.png)
![plotd](https://user-images.githubusercontent.com/87823051/129391480-cd795255-a3b8-4cce-b099-c03b9f66f471.png)
![plotc](https://user-images.githubusercontent.com/87823051/129391479-2844fde5-f802-4036-bdaa-c115f265b4ab.png)

Yee solver is in accordance with analytical results.
Lehe solver shows more y-momentum (16 vs 12.5) making the particle to reach further away (6.5 microns vs 5 microns)
x- momentum and amplitude in both metods are according to analytical result.
The only differences between both simulations' .param and .cfg are this
![Capturalehe](https://user-images.githubusercontent.com/87823051/129393050-43e5c2d9-43f2-4289-914a-db91726a461b.PNG)
Note that in both the particle starts at the same position in y, and the the different initial x position should matter as it is a plane wave.
Thanks

**Comment by sbastrakov** [comment]
Hello @tomaslevyy , could you attach the whole set of .param files for this simulation?

**Comment by sbastrakov** [comment]
I suspect you are generating the field with what picongpu calls `lsser`. However, it is only fully accurate for the standard `Yee` solver (regardless of absorbers). Then what you observe may be caused just by this fact. Now thinking of that, we should probably print a warning for such a combination at runtime. @steindev what do you think?

You have two ways of fully accurate laser setup with other field solvers:

- If you have explicit analytical expressions for `E(x, y, z, t)` and `B(x, y, z, t)`, can set it as field background 
- Make a simulation with Yee that only generates laser before it hits you particle (extend the grid if necessary) and save it in a checkpoint. Then load it and continue with `Lehe`.

**Comment by tomaslevyy** [comment]
Those are the .param

[density.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/6986304/density.txt)
[dimension.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/6986305/dimension.txt)
[fieldAbsorber.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/6986306/fieldAbsorber.txt)
[fieldSolver.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/6986307/fieldSolver.txt)
[fileOutput.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/6986308/fileOutput.txt)
[grid.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/6986309/grid.txt)
[laser.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/6986310/laser.txt)
[memory.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/6986311/memory.txt)
[particle.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/6986312/particle.txt)
[speciesDefinition.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/6986313/speciesDefinition.txt)
[speciesInitialization.txt](https://github.com/ComputationalRadiationPhysics/picongpu/files/6986314/speciesInitialization.txt)

Thanks!

**Comment by sbastrakov** [comment]
Yes, so you are comparing two simulations with slightly different lasers.

Note that of course the two field solvers in question would propagate the laser differently. And you are probably interested to see this effect for your test case.

However, here the laser itself is generated differently due to quirks of our internal implementation. With my suggestions above, you could eliminate this difference.

**Comment by tomaslevyy** [comment]
Perfect, thank you @sbastrakov

**Comment by PrometheusPi** [comment]
@sbastrakov Could you please briefly comment on what differences (in the laser parameters) are to be expected if we use our laser implementation (based on the Yee solver) and use the Lehe solver instead?

**Comment by sbastrakov** [comment]
Tbh I am not sure how to evaluate it. It would just generate a somewhat similar, but different laser.
